meetings/2018-01-17/minutes.md

```diff
@@ -0,0 +1,417 @@
+meeting. The intention of the meeting is to determine the framework within
+which these decisions can be made, how to avoid the fragmentation that Open
+Source is prone to. He suggests that a successful solution will involve a
+critical mass of people unifying behind a single idea.
+
+Jon reminisces that in nearly fifteen years in the Scala community, Scala has
+never had good tooling, and this has held the language back, but the LSP
+project is an idea which already has momentum and enthusiasm that Scala can
+build upon, and this is a unique opportunity for Scala to have a future which
+includes rich tooling.
+
+He cautions the attendees to be aware of the "sunk-cost fallacy", which he
+describes as the feeling that a solution utilizing work already invested seems
+better than a different approach that reaches the end-goal faster, but does not
+make use of the earlier investment, though the latter is the better solution.
+He is aware that this may require some tooling developers to leave behind work
+they're already invested heavily in, and that this may be both difficult and
+necessary.
+
+Jon is especially glad that Martin Odersky is able to attend the meeting, as
+having his blessing on any proposals to arise from the meeting will be
+important in getting support from elsewhere in the community.
+
+## Existing projects
+
+Jon invites many of the people already involved in Scala tooling to speak about
+projects they're involved in, and notes that this is a very important part of
+the discussion to ensure that everyone involved is brought to a common
+understanding of the existing tooling ecosystem.
+
+### Scalameta & SemanticDB - Eugene Burmako, Twitter
+
+Scalameta has almost 50 open-source developers, and as been funded by EPFL,
+JetBrains, Scala Center and Twitter. Scalameta provides APIs optimised for
+tools, designed to be cross-platform, and to work with multiple compilers. It
+provides a syntactic API and a semantic API. SemanticDB is an interchange
+format for Scalameta based on protocol buffers. A SemanticDB document
+corresponds to a source file. SemanticDB can be generated with a compiler
+plugin. The file contains full details of symbols, and details of how names and
+positions and resolve to symbols.
+
+There's a SemanticDB plugin for Scalac 2.11 and 2.12. Many open-source Scala
+projects have been indexed already, including all of Twitter's code. It enables
+Scalafix, Metadoc, Metals and several internal tools at Twitter (which Eugene
+is not at liberty to talk about). It does not require a running compiler, and
+is very fast, even for large databases. The information can be put into an SQL
+database. As an example, two million lines of code can be indexed into an 800MB
+SQLite database, and offers 10ms access times.
+
+Eugene shared the [slides from his presentation](http://goo.gl/TsFfqC).
+
+### Metals - Eugene Burmako, Twitter
+
+Metals ("scalaMETA Language Server") in an LSP implementation based on
```

Comment:
> Metals is a language server implementation using Scalameta-based tools such as Scalafix, Scalafmt and SemanticDB.

Please don't use `("scalaMETA Language Server")`, this capitalization is not used anywhere.

Reply:
I think the intention was to show where do the letters in "metals" come from. Probably it could be achieved with emphasis: "scala**meta** **l**anguage **s**erver"?

Reply:
bold is good!

Reply:
Yes, @laughedelic is right. I made the change as we had been calling it "scalameta/language-server" everywhere before, which is a much less cool name than "Metals". I'm emboldened by the emboldening suggestion.